middleware/proxy.go

```diff
@@ -98,8 +117,9 @@ type (
 var (
 	// DefaultProxyConfig is the default Proxy middleware config.
 	DefaultProxyConfig = ProxyConfig{
-		Skipper:    DefaultSkipper,
-		ContextKey: "target",
+		Skipper:     DefaultSkipper,
+		ContextKey:  "target",
+		RetryFilter: DefaultProxyRetryFilter,
```

Comment:
these `Default*Config` public global structs are not very good things as you can mutate them anywhere, therefore causing hard to find bugs.

going with in middleware func is more bullet-proof
```go
	if config.ErrorHandler == nil {
		config.ErrorHandler = func(err error, c echo.Context) error {
			return err
		}
	}
```